src/modules/Categories.js

```diff
@@ -0,0 +1,227 @@
+    });
+
+    const promise1 = result.then((res) => { res.text().then((a) => console.log(a)); });
+    await promise1;
+  }
+
+  printComments = (comments) => {
+    comments.then((res) => {
+      res.json().then((data) => {
+        const commentsTitle = this.popupComments.querySelector('.comments-title');
+        commentsTitle.innerHTML = `Commments (${data.length})`;
+        const commentsContainer = this.popupComments.querySelector('.comments-container');
+        commentsContainer.innerHTML = '';
+        let newData;
+        for (let i = 0; i < data.length; i += 1) {
+          newData = document.createElement('li');
+          newData.innerHTML = `${data[i].creation_date} ${data[i].username}: ${data[i].comment}`;
+          commentsContainer.appendChild(newData);
+        }
+      });
+    });
+  }
+
+  countComments = (comments) => {
+    comments.then((res) => {
+      res.json().then((data) => {
+        const commentsTitle = this.popupComments.querySelector('.comments-title');
+        commentsTitle.innerHTML = `Commments (${data.length})`;
+        const commentsContainer = this.popupComments.querySelector('.comments-container');
+        commentsContainer.innerHTML = '';
+        let newData;
+        for (let i = 0; i < data.length; i += 1) {
+          newData = document.createElement('li');
+          newData.innerHTML = `${data[i].creation_date} ${data[i].username}: ${data[i].comment}`;
+          commentsContainer.appendChild(newData);
+        }
+      });
+    });
+  }
+
+  getItemApiMain = async (mealId) => {
+    const url = `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${mealId}`;
+    const result = fetch(url);
+
+    await result.then((response) => response.json()).then((json) => {
+      console.log(json.meals[0]);
+
+      const popupImg = this.popupComments.querySelector('#popupImage');
+      popupImg.src = json.meals[0].strMealThumb;
+
+      const popupTitle = this.popupComments.querySelector('#popupTitle');
+      popupTitle.innerHTML = json.meals[0].strMeal;
+
+      const gridContainer = this.popupComments.querySelector('.grid-container');
+      gridContainer.innerHTML = '';
+      let newData = document.createElement('div');
+      newData.innerHTML = `Area:${json.meals[0].strArea}`;
+      gridContainer.appendChild(newData);
+
+      newData = document.createElement('div');
+      newData.innerHTML = `Category:${json.meals[0].strCategory}`;
+      gridContainer.appendChild(newData);
+
+      const ingredientContainer = document.createElement('ul');
+      ingredientContainer.className = 'ingredient-container';
+      gridContainer.appendChild(ingredientContainer);
+      ingredientContainer.innerHTML = '';
+
+      newData = document.createElement('li');
+      newData.innerHTML = 'Ingredients:';
+      ingredientContainer.appendChild(newData);
+
+      const strIngredient = [];
+      for (let i = 1; i <= 20; i += 1) {
+        strIngredient[i] = json.meals[0][`strIngredient${i.toString()}`];
```

Comment:
Consider using strIngredient.push() instead of saving elements in undeclared indexes.